sdk/src/vote_program.rs

```diff
@@ -259,6 +318,97 @@ mod tests {
         let vote = Vote::new(1);
         let vote_state = vote_and_deserialize(&vote_id, &mut vote_account, vote.clone()).unwrap();
         assert_eq!(vote_state.node_id, Pubkey::default());
-        assert_eq!(vote_state.votes, vec![vote]);
+        assert_eq!(vote_state.votes, vec![Lockout::new(&vote)]);
+    }
+
+    #[test]
+    fn test_vote_lockout() {
+        let voter_id = Keypair::new().pubkey();
+        let staker_id = Keypair::new().pubkey();
+        let mut vote_state = VoteState::new(voter_id, staker_id);
+
+        for i in 0..MAX_VOTE_HISTORY {
+            vote_state.process_vote(Vote::new((INITIAL_LOCKOUT as usize * i) as u64));
+        }
+
+        // The last vote should have been popped b/c it reached a depth of MAX_VOTE_HISTORY
+        assert_eq!(vote_state.votes.len(), MAX_VOTE_HISTORY - 1);
+        assert_eq!(vote_state.root_block, Some(0));
+        for (i, vote) in vote_state.votes.iter().enumerate() {
```

Comment:
Is it possible to move this into a utility function and use it in the test below?